### PR TITLE
Undestructive refactoring

### PR DESCRIPTION
**Pull request by MonsieurMan**
Hello there !  
I began using your library for a project I decided to write my first client with electron and it saved me hours of struggle.  I guessed I would put them back in the library, so this is my first PR but expect some more to come as well as issues.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     ],
     "no-string-literal": false,
     "no-console": [
-        false
-    ]
+      false
+    ],
+    "member-ordering": false
```
Comment (dolanmiu) [inline tslint.json:38]: Could you explain more in detail?

I like member-ordering a lot, I think it makes readng code much easier when everything is in alphabetical order

**Comment by dolanmiu** [comment]
Thanks! 

This is overall pretty good

Could you have a look at that comment I put in 87b90b2?

Then I will merge

**Comment by MonsieurMan** [comment]
Hey, I only have my mobile right now and I can't see the comment, have you posted it or is it I just can't find it here with my mobile ?

**Comment by dolanmiu** [comment]
Sorry, forgot to press submit

File: tslint.json (hunk 1 above)
Comment (MonsieurMan) [reply tslint.json:38]: I was like you at first and I changed my mind after reading "*Clean Code*", it's pretty much only for files like the current task/build.ts where I have slit public methods in multiple private ones.  
It makes more sense having the public method above and the private one it uses just below, also they are not used anywhere else in the class. so yes, it is about grouping things that works together to make a group / intention easier to read.  
Anyway it is your project, coding standards are up to you.

**Comment by MonsieurMan** [comment]
By the way, a friend and I are in the process of writing a CLI framework in typescript using an angular like aspect oriented paradigm.  

Basicallly this :
```ts
@Command({
   name: 'example',
   aliases: [ ]
})
export class ExampleCommand implements Executable {
   execute(
      @Param() param: string
   ) { }
}
```  

That's the basic idea, provide compile time typechecking and runtime parsing. So tell me if you're interested in collaborating or a rewrite of this project once we do first stable release.

File: tslint.json (hunk 1 above)
Comment (dolanmiu) [reply tslint.json:38]: hi5! I am reading Clean Code at the moment, uncle bob is my man

Comment (dolanmiu) [reply tslint.json:38]: Very true, you have a point, 

I find it more of a pain if the private method it uses is at the very end of the file, when the public one is near the top

I suppose the rationale is that you shouldn't care of need to delve into the private method, and that it should just be an implementation detail, thats why all private methods can be grouped together

I'll merge this in now, and see how it goes

**Comment by dolanmiu** [comment]
@MonsieurMan 

A CLI for Electron or in general? 

I think this could be interesting. But there is a lot to think about in such an app

`yargs` is really good

**Comment by MonsieurMan** [comment]
A node CLI framework for nodejs in general.  

The thing with all the actual cli frameworks for node is that IMO, functional command module declaration like this makes no sense. (I mean, we're basically forced to declare them all in a single file or find some way to make the constructor that is also the cli module move around and add commands to it).  

Also, I think argument validation decoupled from the actual execution handler is a must have to make the code clearer.

I'll keep you updated and I'll maybe proceed to a rewrite of this project with it once we do first release if it seems good to you. It's [here](https://github.com/orbital-js/orbital) if you want to check it out. I just need to set coding standards because the other contributor is writting really strange code, haha.  

Anyway, thank you for all the time you saved me ! 😄

**Comment by dolanmiu** [comment]
Released as v1.2.3

**Pull request merged**